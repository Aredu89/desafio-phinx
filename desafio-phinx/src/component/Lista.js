import React from 'react';
import './Lista.css';

//Componentes
import Spinner from './common/Spinner.js'
import Error from './common/Error.js'
import Card from './common/Card.js'

const Lista = props =>{
  const { data, isLoading, isError, selectCharacter, setPersonajeFavorito } = props
  return(
    <div className="lista row">
      {isLoading && <Spinner />}
      {isError && <Error message="Error al cargar la lista" />}
      {
        data.length > 0 ? (
          data.map((character, i)=>{
            return <div key={i} className="col-12 col-sm-6 col-md-4 col-lg-3 mb-4">
              <Card character={character} 
                selectCharacter={selectCharacter}
                setPersonajeFavorito={setPersonajeFavorito}
                />
            </div>
          })
        ) : (
          !isLoading && !isError &&
          <div className="w-100 text-center">No hay resultados...</div>
        )
      }
    </div>
  )
}

export default Lista;