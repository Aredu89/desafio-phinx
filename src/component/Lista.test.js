import React from 'react';
import { shallow } from 'enzyme';
import Lista from './Lista.js';

describe('Lista', () => {
  it('Debería renderizar sin datos',()=>{
    const props = {
      data: []
    }
    const component = shallow(<Lista {...props} />);
    expect(component).toMatchSnapshot();
  })
  it('Debería listar 2 cards',()=>{
    const props = {
      data: [
        {
          id: 1,
          name: "1"
        },
        {
          id: 2,
          name: "2"
        },
      ]
    }
    const component = shallow(<Lista {...props} />);
    expect(component
      .find('Card')
      .length).toEqual(2)
  })
  it('Debería mostrar Spinner',()=>{
    const props = {
      data: [],
      isLoading: true
    }
    const component = shallow(<Lista {...props} />);
    expect(component
      .find('Spinner')
      .length).toEqual(1)
  })
  it('Debería mostrar error',()=>{
    const props = {
      data: [],
      isError: true
    }
    const component = shallow(<Lista {...props} />);
    expect(component
      .find('Error')
      .length).toEqual(1)
  })
})