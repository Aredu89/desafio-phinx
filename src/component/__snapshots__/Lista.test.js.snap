// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Lista Deber√≠a renderizar sin datos 1`] = `
<div
  className="lista row"
>
  <div
    className="w-100 text-center"
  >
    No hay resultados...
  </div>
</div>
`;
